### ISSUE TITLE
ts-eslint: AbilityBuilder unbound-method error

### ISSUE THREAD

**Issue by PiotrJakubowski** [stalniy/casl#736]
**Describe the bug**
When destructuring `AbilityBuilder` object I get [@typescript-eslint/unbound-method](https://typescript-eslint.io/rules/unbound-method/) error.

**To Reproduce**
1. Ability configuration
```
import { PureAbility } from '@casl/ability';

type Action = 'manage' | 'create' | 'read' | 'update' | 'delete';
type Subject = 'SomeSettings' | 'all';
type AppAbility = PureAbility<[Action, Subject]>;

const PureAppAbility = PureAbility as AbilityClass<AppAbility>;

export const buildAbilityFor = (): AppAbility => {
// the error's on can and build \/
  const { can, build } = new AbilityBuilder<AppAbility>(PureAppAbility);

// ...
  return build();
};
```
3. How do you check abilities
N/A

**Expected behavior**
Error should not appear. All methods should probably be declared as arrow functions.

**Interactive example** (optional, but highly desirable)
-
**CASL Version**

`@casl/ability` - 6.3.3
`@casl/react` - 3.1.0

**Environment**:
Nodejs 16.13.1 
TypeScript 4.9.4

**Comment by stalniy** [comment]
Hey,

Thanks for the issue. methods are bound using `.bind` call. So, this is not a bug eslint related enhancement. PR is very welcome

**Comment by stalniy** [comment]
After looking in the code, I see it’s not trivial ) 

there is complicated method overload type which I needs to be moved into separate overloaded type and assigned to properties. Need to check whether it doesn’t cause type loosing

**Comment by stalniy** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/636

For reference

**Comment by stalniy** [comment]
fixed in 6.3.4

**Comment by yusufkandemir** [comment]
For anyone interested, this change broke `jest.spyOn(AbilityBuilder.prototype, 'can')`. You need to use alternative ways of mocking with Jest. https://jestjs.io/docs/es6-class-mocks
